### PR TITLE
Geolocation

### PR DESCRIPTION
**Pull request by chasulin**
Geolocates user using [IP-API](https://ip-api.com/). If successful, ranks user's regions at the top of the search bar dropdown menu

### CODE REVIEW

File: src/common/regions/utils.ts [hunk 1]
```diff
@@ -56,3 +60,77 @@ export function getAutocompleteRegions(region?: Region) {
...
+
+// TODO (Chelsi): fix the any
+export function getMetroRegionFromZipCode(zipCode: string): any {
+  const metroFromZip = find(regions.metroAreas, (region: MetroArea) =>
+    find(region.counties, (region: County) =>
+      countyIncludesZip(region, zipCode),
+    ),
+  );
+  return metroFromZip;
+}
```
Comment (pnavarrc) [inline src/common/regions/utils.ts:95]: Consider finding the county first using your function from above, and then use that to find the metro area.

File: src/common/regions/utils.ts [hunk 2]
```diff
@@ -56,3 +60,77 @@ export function getAutocompleteRegions(region?: Region) {
...
+
+// TODO (chelsi): move elsewhere?
+interface GeolocatedRegions {
+  county?: Region;
+  metroArea?: Region;
+  state?: Region;
+}
```
Comment (pnavarrc) [inline src/common/regions/utils.ts:102]: Maybe we don't need a type for this? we could just return an array of `Region` in order

File: src/common/regions/utils.ts [hunk 3]
```diff
@@ -56,3 +60,77 @@ export function getAutocompleteRegions(region?: Region) {
...
+
+/**
+ * If we are able to geolocate the user, we rank their regions (state, county, metro if applicable) first
+ * in the searchbar dropdown menu. This sorts the Region[] accordingly.
+ */
+export function getAutocompleteRegionsWithGeolocation(
+  geolocation: GeolocationInfo,
+  locations: Region[],
+): Region[] {
+  const geolocatedRegions = getGeolocatedRegions(geolocation);
```
Comment (pnavarrc) [inline src/common/regions/utils.ts:126]: Returning an array of `Region` object (with 0, 1, 2, or 3 items) will eliminate the need of sanitizing

**Review by pnavarrc — COMMENTED** [review]
Added some suggestions, will take a second pass tomorrow morning!

File: src/common/regions/utils.ts (hunk 2 above)
Comment (chasulin) [reply src/common/regions/utils.ts:101]: Per our chat, gonna keep it in obj format

**Review by pnavarrc — APPROVED** [review]
Looks good from my location!

**Pull request merged**